paddle/fluid/pir/transforms/onednn/cpu_bfloat16_placement_pass.cc

```diff
@@ -0,0 +1,594 @@
+      return false;
+    }
+
+    for (uint32_t i = 0; i < op->num_results(); i++) {
+      if (!op->result(i) || !op->result(i).type()) {
+        continue;
+      }
+      auto next_op_list = pir::GetUseOpsForOutput(op, i);
+      for (auto const &[next_op, op_index] : next_op_list) {
+        // Some ops do not need to be processed
+        std::string next_op_name = next_op->name();
+        if (constant_op.count(next_op_name)) {
+          continue;
+        }
+        auto op_next_attr = next_op->attributes();
+        if (op_next_attr.find("mkldnn_data_type") == op_next_attr.end()) {
+          // data_type_is_missing
+          VLOG(0) << "data_type_is_missing:" << next_op->name();
+          next_fp32 = true;
+          break;
+        }
+        if (op_next_attr.find("mkldnn_data_type")->second == "float32") {
+          VLOG(0) << "mkldnn_data_type is fp32:" << next_op->name();
+          next_fp32 = true;
+          break;
+        }
+      }
+    }
+
+    if (prev_fp32 && next_fp32) {
+      VLOG(0) << "RemoveOrphanedOperators:" << op->name();
+    } else {
+      VLOG(0) << "Not RemoveOrphanedOperators:" << op->name();
+      return false;
+    }
+
+    std::string target_op_name = op->name();
+    auto op_info =
+        pir::IrContext::Instance()->GetRegisteredOpInfo(target_op_name);
+    if (op_info) {
+      std::vector<pir::Type> op_item_inner_output_types;
+      for (size_t i = 0; i < op->num_results(); ++i) {
+        op_item_inner_output_types.push_back(op->result_type(i));
+      }
+      auto attributes = op->attributes();
+
+      if (attributes.find("mkldnn_data_type") != attributes.end()) {
+        attributes["mkldnn_data_type"] =
+            pir::StrAttribute::get(pir::IrContext::Instance(), "float32");
+      }
+
+      pir::Operation *op_item_inner = rewriter.Build(op->operands_source(),
+                                                     attributes,
+                                                     op_item_inner_output_types,
+                                                     op_info);
+      rewriter.ReplaceOp(op, op_item_inner->results());
+    }
+
+    return true;
+  }
+};
+
+template <typename OpType>
+class RemoveUnsupportedOpPattern : public pir::OpRewritePattern<OpType> {
+ public:
+  using pir::OpRewritePattern<OpType>::OpRewritePattern;
+
+  bool MatchAndRewrite(
+      OpType op,
+      pir::PatternRewriter &rewriter) const override {  // NOLINT
+
+    uint32_t num_operands = op->num_operands();
+    for (uint32_t i = 0; i < num_operands; i++) {
+      auto *pre_op = pir::GetDefiningOpForInput(op, i);
+      if (pre_op->HasAttribute("mkldnn_data_type")) {
+        return false;
+      }
+    }
+
+    bool unsupported_op = false;
+    for (auto &value : op->operands_source()) {
+      pir::Type op_dtype = pir::GetDataTypeFromValue(value);
+      // Only float input can be converted to bfloat16
+      if (!op_dtype.isa<pir::Float32Type>()) {
+        unsupported_op = true;
```

Comment:
So does it means any value having `float32` dtype will enable the pattern? If so then directly `break` after here can realize the target and reduce the loop count.